g4me/analysis/PrepareTracksForMatchingAndFit_embedding.C

```diff
@@ -0,0 +1,330 @@
+
+	          idTrackMID.emplace_back(trackletID);
+
+	          nPreparedTrackletsMID++;
+
+	      }
+      }
+    }
+
+    // end of loop over pythia tree
+    //--------------------------------------------------------------------------
+    // Loop over muon hits
+    for (int iHit=0; iHit<io_signal.hits.n; iHit++) {
+
+      // filling arrays of hit IDs from MID layers (coming from any charged tracks)
+
+      auto trackID = io_signal.hits.trkid[iHit];
+
+      if (!(IsTrackCharged(&io_signal, trackID))) continue;
+
+      mom.SetXYZ(io_signal.hits.px[iHit],io_signal.hits.py[iHit],io_signal.hits.pz[iHit]);
+      if (mom.Mag() < hitMinP) continue;
+
+      if (io_signal.hits.lyrid[iHit] == idLayerMID1) arrayHitID_MIDLayer1[nHits_MIDLayer1++] = iHit;
+      if (io_signal.hits.lyrid[iHit] == idLayerMID2) arrayHitID_MIDLayer2[nHits_MIDLayer2++] = iHit;
+
+      // filling arrays of hits from ITS tracks (only for interesting tracks: charged and primary).
+      // Hits from ITS are by definition all the hits having radius < rMaxITS
+      if (!(IsTrackInteresting(&io_signal, trackID))) continue;
+
+      pos.SetXYZ(gRandom->Gaus(io_signal.hits.x[iHit],resolutionITS),gRandom->Gaus(io_signal.hits.y[iHit],resolutionITS),gRandom->Gaus(io_signal.hits.z[iHit],resolutionITS));
+      if (pos.Perp() < rMaxITS) {
+	       new ((allTracksHitPosITS.at(trackID))[(allTracksHitPosITS.at(trackID)).GetEntries()]) TVector3(pos);
+	       new ((allTracksHitCovITS.at(trackID))[(allTracksHitCovITS.at(trackID)).GetEntries()]) TMatrixDSym(covITS);
+      }
+    }
+
+    // filling the final arrays with the hit information from good ITS tracks
+
+    for (int iTrack=0; iTrack<io_signal.tracks.n; iTrack++) {
+      if (IsTrackInteresting(&io_signal, iTrack)) {
+	       new (trackCandidatesHitPosITS[nPreparedTracksITS]) TClonesArray(allTracksHitPosITS.at(iTrack));
+	       new (trackCandidatesHitCovITS[nPreparedTracksITS]) TClonesArray(allTracksHitCovITS.at(iTrack));
+	       idTrackITS.emplace_back(iTrack);
+	       TParticle part;
+	       part.SetPdgCode(io_signal.tracks.pdg[iTrack]);
+	       part.SetProductionVertex(io_signal.tracks.vx[iTrack],io_signal.tracks.vy[iTrack],io_signal.tracks.vz[iTrack],io_signal.tracks.vt[iTrack]);
+	       part.SetMomentum(io_signal.tracks.px[iTrack],io_signal.tracks.py[iTrack],io_signal.tracks.pz[iTrack],0.001*io_signal.tracks.e[iTrack]);
+         new (particlesITS[nPreparedTracksITS]) TParticle(part);
+         nPreparedTracksITS++;
+      }
+    }
+
+    // filling the final arrays with the hit information from selected MID tracklets
+
+    for (int iHitLayer1=0; iHitLayer1<nHits_MIDLayer1; iHitLayer1++) {
+
+      idHitLayer1 = arrayHitID_MIDLayer1[iHitLayer1];
+      posHitMID1.SetXYZ(gRandom->Gaus(io_signal.hits.x[idHitLayer1],resolutionMID),gRandom->Gaus(io_signal.hits.y[idHitLayer1],resolutionMID),gRandom->Gaus(io_signal.hits.z[idHitLayer1],resolutionMID));
+      trackIdHitLayer1 = io_signal.hits.trkid[idHitLayer1];
```

Comment:
trackIdHitLayer1 and trackIdHitLayer2 should be shifted by io_underlying.tracks.n